sw/device/tests/sim_dv/pwrmgr_deep_sleep_all_wake_ups.c

```diff
@@ -0,0 +1,294 @@
+      .filter = kDifAdcCtrlFilter5,
+      .min_voltage = 0,
+      .max_voltage = 200,
+      .in_range = true,
+      .generate_wakeup_on_match = true,
+      .generate_irq_on_match = false,
+  };
+
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(dif, kDifAdcCtrlChannel0,
+                                             filter_cfg, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(dif, kDifAdcCtrlChannel1,
+                                             filter_cfg, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_adc_ctrl_filter_match_wakeup_set_enabled(
+      dif, kDifAdcCtrlFilter5, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_adc_ctrl_set_enabled(dif, kDifToggleEnabled));
+}
+
+/**
+ * pinmux config for test #3
+ * . use IOB7 as an input
+ * . set posedge detection
+ */
+static void prgm_pinmux_wakeup(void *dif) {
+  dif_pinmux_wakeup_config_t detector_cfg = {
+      .signal_filter = kDifToggleDisabled,
+      .pad_type = kDifPinmuxPadKindMio,
+      .pad_select = kTopEarlgreyPinmuxInselIob7,
+      .mode = kDifPinmuxWakeupModePositiveEdge,
+      .counter_threshold = 0 /* Don't need for posedge detection */,
+  };
+  CHECK_DIF_OK(dif_pinmux_wakeup_detector_enable(dif, kPinmuxWkupDetector5,
+                                                 detector_cfg));
+}
+
+static const test_dev_t kTestDevs[] = {
+    {
+        .name = "SYSRST_CTRL",
+        .base = TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR,
+        .dif = &sysrst_ctrl,
+        .init = sysrst_ctrl_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceOne,
+        .config = prgm_sysrst_ctrl_wakeup,
+    },
+    {
+        .name = "ADC_CTRL",
+        .base = TOP_EARLGREY_ADC_CTRL_AON_BASE_ADDR,
+        .dif = &adc_ctrl,
+        .init = adc_ctrl_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceTwo,
+        .config = prgm_adc_ctrl_wakeup,
+    },
+    {
+        .name = "PINMUX",
+        .base = TOP_EARLGREY_PINMUX_AON_BASE_ADDR,
+        .dif = &pinmux,
+        .init = pinmux_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceThree,
+        .config = prgm_pinmux_wakeup,
+    },
+    {
+        .name = "PWRMGR",
+        .base = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
+        .dif = &pwrmgr,
+        .init = pwrmgr_init,
+        .wakeup_src = (dif_pwrmgr_request_sources_t)kNoWkupDev,
+    },
+    {
+        .name = "RV_PLIC",
+        .base = TOP_EARLGREY_RV_PLIC_BASE_ADDR,
+        .dif = &rv_plic,
+        .init = rv_plic_init,
+        .wakeup_src = (dif_pwrmgr_request_sources_t)kNoWkupDev,
+    },
+};
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  dif_rv_plic_irq_id_t irq_id;
+  top_earlgrey_plic_peripheral_t peripheral;
+
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&rv_plic, kTopEarlgreyPlicTargetIbex0, &irq_id));
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  LOG_INFO("I GOT IRQ %d from peripheral:%d", irq_id, peripheral);
+
```

Comment:
Do you also need to ack and complete the IRQ? ... moreover, can any of the autogenerated ISRs here help your: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/autogen/isr_testutils.c?q=isr_testutils&ss=opentitan%2Fopentitan

Reply:
fxd